website/docs/learning-and-resources/how-to-guides/how-to-work-with-local-apis-on-appsmith/README.mdx

```diff
@@ -0,0 +1,223 @@
+
+
+<TabItem label="localhost" value="localhost">
+
+
+### Datasource on localhost
+
+This guide describes how to connect your self-hosted Appsmith instance to your localhost datasource.
+
+<VideoEmbed host="youtube" videoId="4XlgsVekzhI" title="Connecting from Appsmith in Docker to local datasource" caption="Connecting from Appsmith in Docker to local datasource"/>
+
+To connect to your database on localhost:
+
+1. Ensure that all of the relevant Docker containers are running. If you are running on Linux, [see the note below](#linux-note) before starting. 
+
+2. In your Appsmith app, create a [datasource](/core-concepts/connecting-to-data-sources) for the appropriate type of database or API.
+
+3. In the **Host Address** or URL field where you'd normally provide the URL for your datasource, instead enter `host.docker.internal`.
+
+4. Click **Test** to verify that the connection is valid.
+
+<!-- anchor placed here to accommodate docs page header bar -->
+<a name="linux-note"></a>
+
+Your localhost application should now be connected and query-able from within your Appsmith app.
+
+:::caution note
+**Linux**: If your containers are running on a Linux machine, start your Appsmith container with the following command line argument to `docker run`. This is only supported for Docker 20.10.0 and later.
+
+```bash
+--add-host=host.docker.internal:host-gateway
+```
+
+Or, for `docker-compose`, update your docker-compose.yaml file to include the following `extra_hosts` option:
+
+```yml
+services: ...
+  postgres: ...
+    environment: ...
+    extra_hosts:
+      - "host.docker.internal:host-gateway"
+```
+:::
+
+<figure>
+  <img src="/img/local_db_config.png" style={{width: "100%", height: "auto"}} alt="Connecting from Appsmith in Docker container to PostgreSQL database running on localhost" />
+  <figcaption align="center"><i>Connecting from Appsmith in Docker container to PostgreSQL database running on localhost</i></figcaption>
+</figure>
+
+
+</TabItem>
+
+<TabItem label="Docker network" value="docker-network">
+
+
+### Datasource in Docker
+
+This guide describes how to connect your APIs and databases to your Appsmith instance when they are each in separate Docker containers.
+
+#### Set up services
+
+In this guide, you'll use networked Docker containers to spin up a Postgres instance alongside Appsmith and connect to the database from an Appsmith app.
+
+1. From the command line, create a Docker network with the following:
+
+    ```bash
+    docker network create appsmithnetwork
+    ```
+
+1. Create two folders, `Appsmith` and `Datasource`.
+
+1. Within the `Appsmith` folder, create a `docker-compose.yml` file for Appsmith. Assign it to the network from the first step, such as in the bottom of this example file:
+
+    ```yaml
+    version: "3"
+    services:
+        appsmith:
+            # appsmith-ce for community edition, appsmith-ee for business.
+            image: index.docker.io/appsmith/appsmith-ce:v1.9.12
```

Comment:
I think it would be better to change the version of docker-compose to a newer version or leave it as `latest` here.